core/src/main/java/io/neow3j/protocol/core/methods/response/StackItem.java

```diff
@@ -0,0 +1,187 @@
+    protected Object value;
+
+    public StackItem(StackItemType type, Object value) {
+        this.type = type;
+        this.value = value;
+    }
+
+    public StackItemType getType() {
+        return type;
+    }
+
+    /**
+     * Gets the value of this stack item.
+     *
+     * @return the value of this stack item.
+     */
+    public Object getValue() {
+        return this.value;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        StackItem other = (StackItem) o;
+        return this.type == other.type && Objects.equals(this.value, other.value);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(this.type, this.value);
+    }
+
+    @Override
+    public String toString() {
+        return "StackItem{" +
+                "type=" + this.type +
+                ", value=" + this.value +
+                '}';
+    }
+
+    public ByteArrayStackItem asByteArray() {
+        if (this instanceof ByteArrayStackItem) {
+            return (ByteArrayStackItem) this;
+        }
+        throw new IllegalStateException("This stack item is not of type " +
+                StackItemType.BYTE_ARRAY.jsonValue() + " but of " + this.type.jsonValue());
+    }
+
+    public BooleanStackItem asBoolean() {
+        if (this instanceof BooleanStackItem) {
+            return (BooleanStackItem) this;
+        }
+        throw new IllegalStateException("This stack item is not of type " +
+                StackItemType.BOOLEAN.jsonValue() + " but of " + this.type.jsonValue());
+    }
+
+    public IntegerStackItem asInteger() {
+        if (this instanceof IntegerStackItem) {
+            return (IntegerStackItem) this;
+        }
+        throw new IllegalStateException("This stack item is not of type " +
+                StackItemType.INTEGER.jsonValue() + " but of " + this.type.jsonValue());
+    }
+
+    public ArrayStackItem asArray() {
+        if (this instanceof ArrayStackItem) {
+            return (ArrayStackItem) this;
+        }
+        throw new IllegalStateException("This stack item is not of type " +
+                StackItemType.ARRAY.jsonValue() + " but of " + this.type.jsonValue());
+    }
+
+    public MapStackItem asMap() {
+        if (this instanceof MapStackItem) {
+            return (MapStackItem) this;
+        }
+        throw new IllegalStateException("This stack item is not of type " +
+                StackItemType.MAP.jsonValue() + " but of " + this.type.jsonValue());
+    }
+
+    public StructStackItem asStruct() {
+        if (this instanceof StructStackItem) {
+            return (StructStackItem) this;
+        }
+        throw new IllegalStateException("This stack item is not of type " +
+                StackItemType.STRUCT.jsonValue() + " but of " + this.type.jsonValue());
+    }
+
+    public static class StackDeserializer extends StdDeserializer<StackItem> {
```

Comment:
I think that documentation on the serialization/deserialization does not make sense. So, leave it. :-)